overlord/ifacestate/udevmonitor/udevmon.go

```diff
@@ -131,21 +157,30 @@ func (m *Monitor) udevEvent(ev *netlink.UEvent) {
 }
 
 func (m *Monitor) addDevice(kobj string, env map[string]string) {
-	di, err := hotplug.NewHotplugDeviceInfo(env)
+	dev, err := hotplug.NewHotplugDeviceInfo(env)
 	if err != nil {
 		return
 	}
+	if _, seen := m.devices[dev.DevicePath()]; seen {
+		return
+	}
+	m.devices[dev.DevicePath()] = true
```

Comment:
If `DevicePath()` is expensive to compute perhaps we could do it only once?

Reply:
Good idea, done.